web/src/api/api.js

```diff
@@ -226,17 +228,38 @@ export const submitJobFile = (params, onProgress) => {
     .then((response) => ({response}))
     .catch((err) => ({err}));
 };
+// inst - is use in test to return canned data
+export const submitIngest = async (jobId, onLocked, onResult, inst = axiosInstance, timeout = 5000) => {
 
-export const submitIngest = (jobId, onResult) => {
-  return axiosInstance
+  const postResponse = await inst
     .post(
       'ingestion/ingest/' + jobId,
       {},
       {
         validateStatus: () => true
       }
     )
-    .then((res) => onResult(res));
+    .then(res => res.data)
+    .catch(res => res.data);
+
+  if (postResponse.jobStatus === 'FAILED' && postResponse.reason === 'locked') {
+    onLocked(postResponse);
+    return;
+  };
+
+  let notDone = true;
+  let getResponse;
+
+  while(notDone) {
+    getResponse = await inst.get('ingestion/ingest/' + postResponse.jobLogId, { validateStatus: () => true }).then(res => res.data);
+    notDone = getResponse.jobStatus === 'INGESTING';
+    if(notDone) await sleep(timeout);
```

Comment:
Does this delay 5 secs between every GET request checking ingestion status? This may slow down the whole process unnecessarily and require users to keep the browser tab open for a while.

What if they close the tab by accident or the internet drops out in the middle of the loop?

Are there any specific reasons for the timeout to be 5 seconds? It took me less than 20ms for the GET request to response.

![image](https://github.com/aodn/nrmn-application/assets/26201635/2aefdeb7-c491-4bd8-bcc7-d356c7a67cad)

Reply:
This seems to be the most important to consider, the ingestion process running in the background now appears to be a long SYNC process simply because of waiting for GET request responses.

Reply:
![image](https://github.com/aodn/nrmn-application/assets/26201635/63378e59-fb9f-4b7d-a048-cb9088a9810b)

I'm having over 800 GET requests after nearly an hour, with no indicators of when this would finish and stuck in this screen:

![image](https://github.com/aodn/nrmn-application/assets/26201635/846ac85f-595a-46fb-b1b9-7252fb24ef8a)

Reply:
My understanding is that the ingestion still happens in the background, potentially reaching "INGESTED" status at the end, and maybe failing too. How can we inform the user that feedback if the scenario above happens?

Reply:
My opinion that asking users to wait for ingestion status check from the GET requests and seeing "waiting" on the frontend for sometime doesn't contribute to the whole ingestion process...

I'd prefer after clicking "submit" button, give them no more wait, instead telling them to come back after some time for the ingestion status. 

When they refresh the page or come back later to see the progress, in the job list page, show the status associated to their jobs.